images.go

```diff
@@ -150,13 +150,16 @@ func (r *imageStore) Load() error {
 	r.idindex = truncindex.NewTruncIndex(idlist)
 	r.byid = ids
 	r.byname = names
-	if needSave {
+	if r.IsReadWrite() && shouldSave {
```

Comment:
I suppose we can return an error - the only case where we should want to immediately save things is when the data we loaded has name collisions, and someone opening it for read-write would be able to fix that.